version: "3.8"

services:
  prince:
    image: prince-archiver
    command: ["python", "-m", "prince_archiver.entrypoints.timestep_generator"]
    environment:
      - DATA_DIR=/data/input
    volumes:
      - ./prince_archiver:/app/prince_archiver
      - ./alembic/:/app/alembic
      - input_data:/data/input

  watcher:
    volumes:
      - ./prince_archiver:/app/prince_archiver
      - ./alembic:/app/alembic
      - input_data:/data/input

  worker:
    environment:
      - AWS_BUCKET_NAME=mycostreams
      - AWS_ACCESS_KEY_ID=aws-access-key-id
      - AWS_SECRET_ACCESS_KEY=aws-access-key-id
      - AWS_ENDPOINT_URL=http://s3:9090
    volumes:
      - ./prince_archiver:/app/prince_archiver
      - ./alembic/:/app/alembic
      - input_data:/data/input
    depends_on:
      s3:
        condition: service_healthy

  s3:
    image: adobe/s3mock:3.1.0
    ports:
      - 9090:9090
      - 9191:9191
    environment:
      - initialBuckets=mycostreams
      - validKmsKeys=arn:aws:kms:eu-central-1:1234567890:key/aws-access-key-id
    healthcheck:
      test: ["CMD", "wget", "-q", "--spider", "http://localhost:9090"]
      interval: 5s
      timeout: 5s
      retries: 5


volumes:
  input_data: